src/main/java/com/lalabrand/ecommerce/auth/RefreshTokenService.java

```diff
@@ -0,0 +1,49 @@
+package com.lalabrand.ecommerce.auth;
+
+import com.lalabrand.ecommerce.user.UserRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class RefreshTokenService {
+    private final RefreshTokenRepository refreshTokenRepository;
+    private final UserRepository userRepository;
+    @Value("${refresh.token.expiration.seconds}")
+    private Long refreshTokenExpirationSec;
+
+    public RefreshTokenService(RefreshTokenRepository refreshTokenRepository, UserRepository userRepository) {
+        this.refreshTokenRepository = refreshTokenRepository;
+        this.userRepository = userRepository;
+    }
+
+    @Transactional
+    public RefreshToken createRefreshToken(String email) {
+        RefreshToken refreshToken = RefreshToken.builder()
+                .user(userRepository.findByEmail(email).get())
+                .token(UUID.randomUUID().toString())
+                .expiryDate(Instant.now().plusSeconds(refreshTokenExpirationSec))
+                .build();
+        return refreshTokenRepository.save(refreshToken);
+    }
+
+    public Optional<RefreshToken> findByToken(String token) {
+        return refreshTokenRepository.findByToken(token);
+    }
+
+    public RefreshToken verifyExpiration(RefreshToken token) {
+        if (isTokenExpired(token)) {
+            refreshTokenRepository.delete(token);
```

Comment:
@denomelchenko As I mentioned before, you have to setup TTL based on `expires_at` so that the DB could remove it automatically.

Reply:
In this project we are using mysql which does not have TTL. There is an event that runs once a day and deletes all tokens that have expires_at already expired